### DISCUSSION TITLE
[Bug]: Azure Endpoint gives HTTP 400 - Unsupported data type error

### DISCUSSION THREAD

**Discussion by regstuff** [danny-avila/LibreChat#1361] [Troubleshooting] [answered]
### What happened?

This is the error message I get when I send any message:
`Something went wrong. Here's the specific error message we encountered: Failed to send message. HTTP 400 - Unsupported data type`

### Steps to Reproduce

Here is the Azure section of my .env file
```
AZURE_API_KEY=redacted
AZURE_OPENAI_MODELS=gpt4-turbo
AZURE_OPENAI_DEFAULT_MODEL=gpt4-turbo
# PLUGINS_USE_AZURE="true"
#AZURE_USE_MODEL_AS_DEPLOYMENT_NAME=TRUE

AZURE_OPENAI_API_INSTANCE_NAME=engpub-openai-switzerland
AZURE_OPENAI_API_DEPLOYMENT_NAME=gpt-4-default
AZURE_OPENAI_API_VERSION=2023-07-01-preview
```
My Azure instance is working. The following curl command returns a proper response
```
curl "https://engpub-openai-switzerland.openai.azure.com/openai/deployments/gpt-4-default/chat/completions?api-version=2023-07-01-preview" \
  -H "Content-Type: application/json" \
  -H "api-key: Redacted" \
  -d "{
  \"messages\": [{\"role\":\"system\",\"content\":\"You are an AI assistant that helps people find information.\"}],
  \"max_tokens\": 800,
  \"temperature\": 0.7,
  \"frequency_penalty\": 0,
  \"presence_penalty\": 0,
  \"top_p\": 0.95,
  \"stop\": null
}"
```
This is my settings pane in Librechat
![image](https://github.com/danny-avila/LibreChat/assets/6016831/96d501d9-619a-43bb-9abe-1df84fc55811)

Not sure if this is a bug or if I'm doing something wrong. This is my first time with LibreChat. OpenAI endpoints work fine btw. 
Great app btw!

### What browsers are you seeing the problem on?

Microsoft Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
Thanks for checking out the project! And for providing env vars.

These values are suspect:
```
AZURE_OPENAI_MODELS=gpt4-turbo
AZURE_OPENAI_DEFAULT_MODEL=gpt4-turbo
```

Azure is working for me: 
![image](https://github.com/danny-avila/LibreChat/assets/110412045/6c1115bf-0047-418c-98d4-01b386cf9d83)

these are the only values I'm using: 
```bash
AZURE_API_KEY=api-key
AZURE_OPENAI_API_INSTANCE_NAME=my_instance
AZURE_OPENAI_API_DEPLOYMENT_NAME=gpt-35-turbo
AZURE_OPENAI_API_VERSION=2023-07-01-preview
```

I'm able to reproduce your issue when providing the AZURE_OPENAI_DEFAULT_MODEL. I'm going to open this back up as an issue and fix this now

Thanks again for your report!

**Reply by danny-avila** [reply]
#1362
